toolbox/arcgee/data.py

```diff
@@ -1054,6 +1054,27 @@ def init_and_set_tags(project=None, workload_tag=None):
     return
 
 
+# Get ROI from the object extent.
+# Use string-based type hints to avoid type checking errors in ArcGIS Pro 3.2.
+def get_roi_from_object(obj: "ee.Image | ee.FeatureCollection") -> "ee.Geometry.BBox":
```

Comment:
What versions of python need to be supported for this library. earthengine_api supports python >= 3.9 and `|` instead of union starts with python 3.10.